autoload/codefmt/formatterhelpers.vim

```diff
@@ -0,0 +1,74 @@
+" Copyright 2020 Google Inc. All rights reserved.
+"
+" Licensed under the Apache License, Version 2.0 (the "License");
+" you may not use this file except in compliance with the License.
+" You may obtain a copy of the License at
+"
+"     http://www.apache.org/licenses/LICENSE-2.0
+"
+" Unless required by applicable law or agreed to in writing, software
+" distributed under the License is distributed on an "AS IS" BASIS,
+" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+" See the License for the specific language governing permissions and
+" limitations under the License.
+
+
+function! s:EnsureIsSyscall(Value) abort
+  if type(a:Value) == type({}) &&
+      \ has_key(a:Value, 'Call') &&
+      \ maktaba#function#HasSameName(
+          \ a:Value.Call, function('maktaba#syscall#Call'))
+    return a:Value
+  endif
+  throw maktaba#error#BadValue(
+      \ 'Not a valid matkaba.Syscall: %s', string(a:Value))
+endfunction
+
+
+""
+" @public
+" Format lines in the current buffer via a formatter invoked by {cmd} (a
+" |maktaba.Syscall|). The command includes the explicit range line numbers to
+" use, if any.
+"
+" @throws ShellError if the {cmd} system call fails
+function! codefmt#formatterhelpers#Format(cmd) abort
+  call s:EnsureIsSyscall(a:cmd)
+  let l:lines = getline(1, line('$'))
+  let l:input = join(l:lines, "\n")
+
+  let l:result = maktaba#syscall#Create(a:cmd).WithStdin(l:input).Call()
```

Comment:
If a:cmd is a Syscall already, you don't need to call `Create()`.

(Though as mentioned elsewhere, maybe accepting either is fine? `Create(Syscall)` is documented to return itself, for convenience.)

Reply:
Resolved.